packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -55,18 +61,76 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
     constructor(
         address _libAddressManager,
         uint256 _maxTransactionGasLimit,
-        uint256 _l2GasDiscountDivisor,
-        uint256 _enqueueGasCost
+        uint256 _enqueueGasCost,
+        uint256 _l2GasDiscountDivisor
     )
         Lib_AddressResolver(_libAddressManager)
     {
         maxTransactionGasLimit = _maxTransactionGasLimit;
-        L2_GAS_DISCOUNT_DIVISOR = _l2GasDiscountDivisor;
-        ENQUEUE_GAS_COST  = _enqueueGasCost;
-        ENQUEUE_L2_GAS_PREPAID = _l2GasDiscountDivisor * _enqueueGasCost;
+        enqueueGasCost  = _enqueueGasCost;
+        l2GasDiscountDivisor = _l2GasDiscountDivisor;
+        enqueueL2GasPrepaid = _l2GasDiscountDivisor * _enqueueGasCost;
     }
 
 
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+
+    /**
+     * Modifier to enforce that, if configured, only the OVM_Sequencer contract may
+     * successfully call a method.
+     */
+    modifier onlySequencer() {
+        require(
+            msg.sender == resolve("OVM_Sequencer"),
```

Comment:
Now that I think about it, it's probably not the sequencer we want here. 
Do we need a new entry in the AddressManager?

Reply:
Yes, we should have a `ConfigManagerAddress`

Reply:
What is that the name of? Would it be a function that should be added to the CTC or AM?
How would the code above change for that, like this?

```suggestion
    modifier onlyManager() {
        require(
            msg.sender == resolve("OVM_Manager"),
```

Reply:
Going to change this to: 
```
    modifier onlyBurnAdmin() {
        require(
            msg.sender == resolve("OVM_BurnAdmin"),
            "Only callable by the BurnAdmin."
        );
        _;
    }
 ```
 
 But won't be able to make the needed changes to the deploy scripts for a few hours.

Reply:
FWIW I would +1 Karl's suggestion to just use `addressManager.owner`.

Reply:
Maybe it's an argument for using the musig less for this non-security-critical param?

Reply:
There's nothing preventing `resolve("OVM_BurnAdmin")` from returning `addressManager.owner` or any other address we choose. 
So I like this approach as it provides the option to use a unique address or reuse another one.